e2e/_suites/ingest-manager/fleet.go

```diff
@@ -299,109 +338,428 @@ func (fts *FleetTestSuite) systemPackageDashboardsAreListedInFleet() error {
+	agentID, err := getAgentID(fts.Hostname)
+	if err != nil {
+		return err
+	}
+
+	maxTimeout := 2 * time.Minute
+	retryCount := 1
+
+	exp := e2e.GetExponentialBackOff(maxTimeout)
+
+	getEventsFn := func() error {
+		listed, err := isPolicyResponseListedInSecurityApp(agentID)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"elapsedTime": exp.GetElapsedTime(),
+				"err":         err,
+				"retries":     retryCount,
+			}).Warn("Could not get metadata from the Administration view in the Security App yet")
+			retryCount++
+
+			return err
+		}
+
+		if !listed {
+			log.WithFields(log.Fields{
+				"agentID":     agentID,
+				"elapsedTime": exp.GetElapsedTime(),
+				"retries":     retryCount,
+			}).Warn("The policy response is not listed as 'success' in the Administration view in the Security App yet")
+			retryCount++
+
+			return fmt.Errorf("The policy response is not listed as 'success' in the Administration view in the Security App yet")
+		}
+
+		log.WithFields(log.Fields{
+			"elapsedTime": exp.GetElapsedTime(),
+			"retries":     retryCount,
+		}).Info("The policy response is listed as 'success' in the Administration view in the Security App")
+		return nil
+	}
+
+	err = backoff.Retry(getEventsFn, exp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (fts *FleetTestSuite) thePolicyIsUpdatedToHaveMode(name string, mode string) error {
+	if name != "malware" {
+		log.WithFields(log.Fields{
+			"name": name,
+		}).Warn("We only support 'malware' policy to be updated")
+		return godog.ErrPending
+	}
+
+	if mode != "detect" && mode != "prevent" {
+		log.WithFields(log.Fields{
+			"name": name,
+			"mode": mode,
+		}).Warn("We only support 'detect' and 'prevent' modes")
+		return godog.ErrPending
+	}
+
+	integration, err := getIntegrationFromAgentPolicy("Elastic Endpoint", fts.PolicyID)
+	if err != nil {
+		return err
+	}
+	fts.Integration = integration
+
+	integrationJSON := fts.Integration.json
+
+	// prune fields not allowed in the API side
+	prunedFields := []string{
+		"created_at", "created_by", "id", "revision", "updated_at", "updated_by",
+	}
+	for _, f := range prunedFields {
+		integrationJSON.Delete(f)
+	}
+
+	// wee only support Windows and Mac, not Linux
+	integrationJSON.SetP(mode, "inputs.0.config.policy.value.windows."+name+".mode")
+	integrationJSON.SetP(mode, "inputs.0.config.policy.value.mac."+name+".mode")
+
+	response, err := updateIntegrationPackageConfig(fts.Integration.packageConfigID, integrationJSON.String())
 	if err != nil {
 		return err
 	}
 
+	success := response.Path("success").Data().(bool)
+	if !success {
+		return fmt.Errorf("The update of the integration package configuration failed. %v", response)
+	}
+
+	// we use a string because we are not able to process what comes in the event, so we will do
+	// an alphabetical order, as they share same layour but different millis and timezone format
+	updatedAt := response.Path("item.updated_at").Data().(string)
+	fts.PolicyUpdatedAt = updatedAt
```

Comment:
We store the moment the policy is updated, so that we can detect events for an agent that happened right after the update. We are storing as string for a reason: events timestamp comes in multiple formats, which makes difficult to parse it with Go layouts (or at least I was not able to parse events' timestamps in a generic way)